Mathlib/Topology/Algebra/OpenSubgroup.lean

```diff
@@ -364,3 +364,91 @@ theorem isOpen_of_isOpen_subideal {U I : Ideal R} (h : U ≤ I) (hU : IsOpen (U
   @Submodule.isOpen_mono R R _ _ _ _ Semiring.toModule _ _ h hU
 
 end Ideal
+
+/-!
+# Open normal subgroups of a topological group
+
+This section builds the lattice `OpenNormalSubgroup G` of open subgroups in a topological group `G`,
+and its additive version `OpenNormalAddSubgroup`.
+-/
+
+section
+
+universe u v
+
+/-- The type of open normal subgroups of a topological group. -/
+@[ext]
+structure OpenNormalSubgroup (G : Type u) [Group G] [TopologicalSpace G]
+  extends OpenSubgroup G where
+  isNormal' : toSubgroup.Normal := by infer_instance
+
+/-- The type of open normal subgroups of a topological additive group. -/
+@[ext]
+structure OpenNormalAddSubgroup (G : Type u) [AddGroup G] [TopologicalSpace G]
+  extends OpenAddSubgroup G where
+  isNormal' : toAddSubgroup.Normal := by infer_instance
+
+attribute [to_additive] OpenNormalSubgroup
+
+namespace OpenNormalSubgroup
+
+variable {G : Type u} [Group G] [TopologicalSpace G]
+
+@[to_additive]
+instance (H : OpenNormalSubgroup G) : H.toSubgroup.Normal := H.isNormal'
+
+@[to_additive]
+theorem toSubgroup_injective : Function.Injective
+    (fun H => H.toOpenSubgroup.toSubgroup : OpenNormalSubgroup G → Subgroup G) :=
+  fun A B h => by
+  ext
+  dsimp at h
+  rw [h]
+
+@[to_additive]
+instance : SetLike (OpenNormalSubgroup G) G where
+  coe U := U.1
+  coe_injective' _ _ h := toSubgroup_injective <| SetLike.ext' h
+
+@[to_additive]
+instance : SubgroupClass (OpenNormalSubgroup G) G where
+  mul_mem := Subsemigroup.mul_mem' _
+  one_mem U := U.one_mem'
+  inv_mem := Subgroup.inv_mem' _
+
+@[to_additive]
+instance : Coe (OpenNormalSubgroup G) (Subgroup G) where
+  coe := fun H => H.toOpenSubgroup.toSubgroup
+
+@[to_additive]
+instance instPartialOrderOpenNormalSubgroup : PartialOrder (OpenNormalSubgroup G) := inferInstance
```

Comment:
Is there a reason that you explicitly name these instances? If not, please remove them. Also below and in the other file.

And in this particular case, it seems that Lean can already find the instance. So is there a reason to add it explicitly?

Reply:
I just copy `OpenSubgroup`, they can both be removed, should I remove the `OpenSubgroup` one too?